### DISCUSSION TITLE
Welcome to isctl Discussions!

### DISCUSSION THREAD

**Discussion by cgascoig** [cgascoig/isctl#68] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.

**Comment by bcassaro** [comment]
@cgascoig 
Hi Chris, thanks to you and the team for writing the isctl tool.  We find it quite useful!

We're seeking the syntax on how to claim an FI pair.  We've reviewed all the help documentation and cannot seem to locate the ability to do so.  Perhaps it does not currently exist?

Any feedback is appreciated.
Thanks,
Bryan

**Reply by cgascoig** [reply]
Hi Bryan,

Thanks for trying isctl! Glad to hear it is useful.

I don't have a spare FI pair handy to fully test this myself right now but the syntax to claim a device with a Device ID and Claim Code should be like this:

```
isctl create asset deviceclaim --SerialNumber <DEVICE_ID> --SecurityToken <CLAIM_CODE>
```

If you try this (or have already tried this) and it doesn't work please reply back here and I will free up some lab equipment to confirm the syntax.

Regards,
Chris

**Comment by bcassaro** [comment]
Thank you Chris!

We are out of the office tomorrow, but will try this as soon as possible!

We're working on automating the bring up and Intersight claim process as much as possible.

Thanks for getting back to me!

Bryan


________________________________

**Comment by bcassaro** [comment]
Hi Chris,

Good news, we we're able to test it out today and found that it works!!!

Do you happen to know the syntax to specify the resource group and organization?

Like I said before, we're attempting to use the isctl along with some other products to automate our infrastructure bring-up.  Thanks for writing this application, we find it extremely useful.


Thank you,
Bryan Cassaro

**Reply by cgascoig** [reply]
To associate a claimed device connector to a resource group (and organisation by extension) at claim time is a two step process. 

First you create a reservation that references the Resource Group(s) like this. Replace <RESOURCE_GROUP> with the resource group name.
```
isctl create resource reservation --ResourceType asset.DeviceRegistration --Groups "[{\"ClassId\": \"mo.MoRef\", \"ObjectType\": \"resource.Group\", \"Selector\": \"Name eq '<RESOURCE_GROUP>'\"}]"
```

Then when you claim the device, you reference the reservation:
```
isctl create asset deviceclaim --SecurityToken 1234 --SerialNumber 1234 --Reservation "MoRef[Moid:6374663d6972652d33af89ea]"
```

If you're on MacOs/Linux you should be able to link those like this:
```
RESERVATION_MOID=$(isctl create resource reservation --ResourceType asset.DeviceRegistration --Groups "[{\"ClassId\": \"mo.MoRef\", \"ObjectType\": \"resource.Group\", \"Selector\": \"Name eq '<RESOURCE_GROUP>'\"}]" -o jsonpath=".Moid")
isctl create asset deviceclaim --SecurityToken 1234 --SerialNumber 1234 --Reservation "MoRef[Moid:${RESERVATION_MOID}]"
```

Couple of notes:
- The examples above all assume a MacOS/Linux shell - for Windows the quoting and escaping is probably different
- The syntax above is a little more complicated than I would like - I'm thinking of some enhancements to make the reservation command a little easier
- Let me know if this does/doesn't work for you (again, I didn't actually test this with spare hardware) so I can capture it as an example in the user guide

**Reply by bcassaro** [reply]
Hi Chris,

I apologize for the delay, but wanted to circle back and thank you for the help with the syntax.  We followed the model for Linux and it works perfectly.  

I thought that I would share back that we're using a 3rd party orchestration product, via this product, we're doing a combination of bash shell scripting, isctl and Terraform (thanks to Tyson Scott) to automate the bring-up of an IMM FI pair into Intersight, claim, deploy and then configure the ports.  Next, discovery the servers (C-Series), then automate the installation of ESXi, finally, automate the installation of VCF (ESXi --> Cloud Builder --> SDDC Manager). 

The isctl CLI tool is very helpful, so thank you!

Bryan
